api/v1beta1/common_types.go

```diff
@@ -117,6 +117,16 @@ const (
 	// MachineSkipRemediationAnnotation is the annotation used to mark the machines that should not be considered for remediation by MachineHealthCheck reconciler.
 	MachineSkipRemediationAnnotation = "cluster.x-k8s.io/skip-remediation"
 
+	// MachineSetSkipPreflightChecksAnnotation is the annotations used to provider a coma separated list of
+	// preflight checks that should be skipped during the MachineSet reconciliation.
+	// Supported items are:
```

Comment:
kubeadm format looks something like this:
* "SystemVerification,Swap,FileContent--proc-sys-net-bridge-bridge-nf-call-iptables"
* "KubernetesVersion", "KubeletVersion"

So I would suggest:
* KubeadmVersion
* KubernetesVersion
* ControlPlaneStable

(I would rename the constants and their values accordingly)

Reply:
@ykakarap Could we use the same kapitalization as kubeadm?

Reply:
Ah okay.  misunderstood your original comment.